src/waveform/renderers/waveformrendermark.h

```diff
@@ -1,20 +1,34 @@
 #ifndef WAVEFORMRENDERMARK_H
 #define WAVEFORMRENDERMARK_H
 
+#include <QObject>
+
 #include "skin/skincontext.h"
 #include "util/class.h"
 #include "waveform/renderers/waveformmarkset.h"
 #include "waveform/renderers/waveformrendererabstract.h"
+#include "library/dao/cue.h"
+#include "preferences/configobject.h"
 
-class WaveformRenderMark : public WaveformRendererAbstract {
+class WaveformRenderMark : public QObject, public WaveformRendererAbstract {
+    Q_OBJECT
   public:
     explicit WaveformRenderMark(WaveformWidgetRenderer* waveformWidgetRenderer);
 
     virtual void setup(const QDomNode& node, const SkinContext& context);
     virtual void draw(QPainter* painter, QPaintEvent* event);
 
+    // Called when a new track is loaded.
+    virtual void onSetTrack();
+
+  public slots:
+    // Called when the loaded track's cues are added, deleted or modified and
+    // when a new track is loaded.
+    // It updates the marks' names and regenerates their image if needed.
+    void slotCuesUpdated();
+
   private:
-    void generateMarkImage(WaveformMark& mark);
+    void generateMarkImage(WaveformMark* mark);
```

Comment:
Just a minor naming issue: Should be pMark in both this declaration and the definition in the .cpp file.